lib/resolvers/http.js

```diff
@@ -58,9 +95,16 @@ module.exports = {
       u.protocol = url.parse(location.href).protocol;
     }
 
-    return fetch(this.makeRequest(url.format(u)))
-      .then((response) => response.arrayBuffer())
-      .then((buffer) => Buffer.from(buffer))
+    const request = new Request(url.format(u), {
+      method: "GET",
+      headers: this.headers,
+      redirect: this.redirects,
+      timeout: this.timeout,
```

Comment:
Unfortunately, `Request` does not support timeout.
You need to use [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).
There is a neat usage example included in the link above.

Reply:
see https://github.com/node-fetch/node-fetch#request-cancellation-with-abortsignal

i think, remove timeout option may be better.

Reply:
if so, need to add extra library, https://npm.im/abort-controller

Reply:
Let's add the library. The fact the last release took place 2 years is nothing wrong. The spec has simply not changed since then, so there was no need to introduce any change.
I had a chance to utilize that library in the past and I can say it's solid.